aws/aws.go

```diff
@@ -36,98 +36,68 @@ const topicNameMaxLength = 40 - len(topicNamePrefix)
 // VerifyPermissions checks if the aws config exists and checks if it has permissions to
 // create sns topics, send messages, create SQS topics, delete topics, and delete sqs queues
 func VerifyPermissions(cfg *aws.Config) error {
-
 	// Check if environment contains aws config
-	_, errConfig := cfg.Credentials.Get()
-	if errConfig != nil {
-		return errConfig
-	}
-
 	topic := fmt.Sprintf("verifyPermissions")
-	subscriptionID := "verifySubsId"
+	subscriptionID := uuid.New().String()
+
+	sess, err := ensureSession(cfg)
+	if err != nil {
+		return err
+	}
 
-	s, subErr := NewSubscriber(cfg, topic, subscriptionID)
+	s, subErr := newSubscriber(sns.New(sess), sqs.New(sess), topic, subscriptionID)
 	if subErr != nil {
-		return errors.New("error, failed to create a SQS queue: " + subErr.Error())
+		return subErr
 	}
 
-	publisher, pubErr := NewPublisher(cfg, topic)
+	publisher, pubErr := newPublisher(sns.New(sess), topic)
 	if pubErr != nil {
-		return errors.New("error, failed to create a SNS topic: " + pubErr.Error())
+		return pubErr
 	}
-
-	return verifyPermissions(s, publisher, topic, subscriptionID)
+	return verifyPermissions(s, publisher)
```

Comment:
nitpicky code style thing: making the variable names consistent would be nice. Call them both `subscriber` and `publisher`